packages/react/runtime/src/hydrate.ts

```diff
@@ -303,7 +303,8 @@ export function hydrate(before: SnapshotInstance, after: SnapshotInstance, optio
         const diffResult = diffArrayLepus(
           beforeChildNodes,
           afterChildNodes,
-          (a, b) => a.type === b.type,
+          (a, b) =>
+            a.type === b.type && a.__listItemPlatformInfo?.['item-key'] === b.__listItemPlatformInfo?.['item-key'],
```

Comment:
@DwwWxx What case of error does this fix, can we add some unit test cases to avoid future regression? Thanks!

Reply:
By the way, this issue can also be solved by https://github.com/lynx-family/lynx-stack/pull/1902, so no need to merge this pr.